sim/rogue/item_sets_phase_8.go

```diff
@@ -1,6 +1,121 @@
+// afflicting the target, up to a maximum increase of 30%
+func (rogue *Rogue) applyScarletEnclaveDamage2PBonus() {
+	if rogue.HasRune(proto.RogueRune_RuneJustAFleshWound) {
+		return
+	}
+
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 2P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	totalBleedsAndPoisons := rogue.PoisonsActive + rogue.BleedsActive
+	spellsModifiedBySetBonus := ClassSpellMask_RogueBackstab | ClassSpellMask_RogueSinisterStrike | ClassSpellMask_RogueSaberSlash | ClassSpellMask_RogueMutilate
+
+	damageMod := rogue.AddDynamicMod(core.SpellModConfig{
+		Kind:       core.SpellMod_DamageDone_Pct,
+		ClassMask:  spellsModifiedBySetBonus,
+		FloatValue: 1.0,
+	})
+
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		ActionID:       core.ActionID{SpellID: 1226843},
+		Name:           label,
+		Callback:       core.CallbackOnApplyEffects,
+		ProcChance:     0.1,
+		ClassSpellMask: spellsModifiedBySetBonus,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			fmt.Println("Bleeds: ", rogue.BleedsActive)
+			fmt.Println("Poisons: ", rogue.PoisonsActive)
+
+			// Only apply the damage mod up to 3 times for the 30% bonus maximum
+			fmt.Println("2P proc triggered")
+			damageMod.UpdateFloatValue(1 + 0.10*float64(min(3, totalBleedsAndPoisons)))
+			damageMod.Activate()
+		},
+	})
+}
+
+// 4P
+// Your Poison and autoattack critical strikes have a 10% chance to grant you a combo point. (Proc chance: 10%)
+func (rogue *Rogue) applyScarletEnclaveDamage4PBonus() {
+	label := "S03 - Item - Scarlet Enclave - Rogue - Damage 4P Bonus"
+	if rogue.HasAura(label) {
+		return
+	}
+
+	// Combo! spell that adds the combo point for the set bonus
+	comboPointMetrics := rogue.NewComboPointMetrics(core.ActionID{SpellID: 1226869})
+
+	// Add a combo point for poison tick crit
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		Name:           label,
+		ClassSpellMask: ClassSpellMask_RogueDeadlyPoisonTick | ClassSpellMask_RogueOccultPoisonTick | ClassSpellMask_RogueInstantPoison,
+		Callback:       core.CallbackOnPeriodicDamageDealt | core.CallbackOnSpellHitDealt,
+		Outcome:        core.OutcomeCrit,
+		ProcChance:     0.10,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
+		},
+	})
+
+	// Add a combo point for autoattack crit | Not sure if I need to add anything here to track autoattacks
+	core.MakeProcTriggerAura(&rogue.Unit, core.ProcTrigger{
+		Name:       label,
+		Callback:   core.CallbackOnPeriodicDamageDealt | core.CallbackOnSpellHitDealt,
+		Outcome:    core.OutcomeCrit,
+		ProcChance: 0.10,
+		Handler: func(sim *core.Simulation, spell *core.Spell, result *core.SpellResult) {
+			rogue.AddComboPoints(sim, 1, rogue.CurrentTarget, comboPointMetrics)
+		},
+	})
```

Comment:
On this proc trigger, in order to make it only proc from auto attacks and not from other things like melee abilities, spell hits, etc, you would want to set the `ProcMask` option to restrict the ProcMasks that can proc it
```
ProcMask: core.ProcMaskMeleeWhiteHit
```